presto-main/src/test/java/com/facebook/presto/execution/scheduler/TestSourcePartitionedScheduler.java

```diff
@@ -189,8 +195,9 @@ public void testScheduleSplitsBatched()
         for (int i = 0; i <= (60 / 7); i++) {
             ScheduleResult scheduleResult = scheduler.schedule();
 
+            assertFalse(scheduleResult.isFinished());
             // finishes when last split is fetched
-            assertEquals(scheduleResult.isFinished(), i == (60 / 7));
+            assertEquals(scheduleResult.getBlockedReason(), i == (60 / 7) ? Optional.of(NO_ACTIVE_DRIVER_GROUP) : Optional.empty());
```

Comment:
I think it would be necessary to introduce `assertFinished(scheduledResult, scheduler)`. It will allow two possible behaviors:

  * finished, not blocked
  * initially: not finished, not blocked; on 2nd invocation of schedule: finished, not blocked

In addition, I believe something is off here. When `i == 60/7`, the code is effectively doing:

```
assertEquals(scheduleResult.getBlockedReason, Optional.of(NO_ACTIVE_DRIVER_GROUP));
assertTrue(scheduleResult.getBlocked().isDone());
```

I think it would be reasonable to introduce `assertBlocked(optionalReason)`